src/Countries/Malaysia.php

```diff
@@ -0,0 +1,540 @@
+        }
+
+        return $variableHolidays;
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionHolidays(int $year): array
+    {
+        if ($this->region && !$this->validRegion($this->region)) {
+            throw InvalidRegion::notFound($this->region);
+        }
+
+        return $this->holidaysByRegion($year);
+    }
+
+    protected function validRegion(string $region): bool
+    {
+        return in_array($region, $this->regions);
+    }
+
+    /**
+     * return holiday by region, if not set return all
+     * 
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function holidaysByRegion(int $year): array
+    {
+        return match ($this->region) {
+            'jhr' => $this->regionJohor($year),
+            'kdh' => $this->regionKedah($year),
+            'ktn' => $this->regionKelantan($year),
+            'kul' => $this->regionKualaLumpur($year),
+            'lbn' => $this->regionLabuan($year),
+            'mlk' => $this->regionMelaka($year),
+            'nsn' => $this->regionNegeri9($year),
+            'phg' => $this->regionPahang($year),
+            'png' => $this->regionPenang($year),
+            'prk' => $this->regionPerak($year),
+            'pls' => $this->regionPerlis($year),
+            'pjy' => $this->regionPutrajaya($year),
+            'sbh' => $this->regionSabah($year),
+            'swk' => $this->regionSarawak($year),
+            'sgr' => $this->regionSelangor($year),
+            'trg' => $this->regionTerengganu($year),
+            default => array_merge(
+                $this->regionJohor($year),
+                $this->regionKedah($year),
+                $this->regionKelantan($year),
+                $this->regionKualaLumpur($year),
+                $this->regionLabuan($year),
+                $this->regionMelaka($year),
+                $this->regionNegeri9($year),
+                $this->regionPahang($year),
+                $this->regionPenang($year),
+                $this->regionPerak($year),
+                $this->regionPerlis($year),
+                $this->regionPutrajaya($year),
+                $this->regionSabah($year),
+                $this->regionSarawak($year),
+                $this->regionSelangor($year),
+                $this->regionTerengganu($year),
+            ),
+        };
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionJohor(int $year): array
+    {
+        $johorHolidays = [
+            'Hari Keputeraan Sultan Johor' => CarbonImmutable::createFromDate($year, 3, 23, $this->timezone),
+            'Hari Thaipusam' => $this->hariThaipusam($year),
+            'Awal Ramadan' => $this->awalRamadan($year),
+            'Hari Deepavali' => $this->hariDeepavali($year),
+        ];
+
+        $hariHolJohor = match ($year) {
+            2020 => CarbonImmutable::createFromDate($year, 9, 24, $this->timezone),
```

Comment:
Timezones shouldn't be needed (this is for all the CarbonImmutable objects using the timezone.